modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/evaluator/aggregator/RegressionMetricStatsAggregator.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.selection.scoring.evaluator.aggregator;
+
+import org.apache.ignite.ml.IgniteModel;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+import org.apache.ignite.ml.selection.scoring.evaluator.context.EmptyContext;
+import org.apache.ignite.ml.structures.LabeledVector;
+
+/**
+ * Class represents statistics aggregator for regression estimation.
+ */
+public class RegressionMetricStatsAggregator implements MetricStatsAggregator<Double, EmptyContext<Double>, RegressionMetricStatsAggregator> {
+    /** Serial version uid. */
+    private static final long serialVersionUID = -2459352313996869235L;
+
+    /**
+     * Number of examples in dataset.
+     */
+    private long N = 0;
+
+    /**
+     * Absolute error.
+     */
+    private double absoluteError = Double.NaN;
+
+    /**
+     * Resudual sum of squares.
+     */
+    private double rss = Double.NaN;
+
+    /**
+     * Sum of labels.
+     */
+    private double sumOfYs = Double.NaN;
+
+    /**
+     * Sum of squared labels.
+     */
+    private double sumOfSquaredYs = Double.NaN;
+
+    /**
+     * Creates an instance of RegressionMetricStatsAggregator.
+     */
+    public RegressionMetricStatsAggregator() {
+    }
+
+    /**
+     * Creates an instance of RegressionMetricStatsAggregator.
+     *
+     * @param n Number of examples in dataset.
+     * @param absoluteError Absolute error.
+     * @param rss Rss.
+     * @param sumOfYs Sum of labels.
+     * @param sumOfSquaredYs Sum of squared labels.
+     */
+    public RegressionMetricStatsAggregator(long n, double absoluteError, double rss, double sumOfYs,
+        double sumOfSquaredYs) {
+        N = n;
+        this.absoluteError = absoluteError;
+        this.rss = rss;
+        this.sumOfYs = sumOfYs;
+        this.sumOfSquaredYs = sumOfSquaredYs;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void aggregate(IgniteModel<Vector, Double> model, LabeledVector<Double> vector) {
+        N += 1;
+        Double prediction = model.predict(vector.features());
+        Double truth = vector.label();
+        double error = truth - prediction;
```

Comment:
What if `prediction` or `truth` is null?

Reply:
lets add assert with comment

Reply:
fixed